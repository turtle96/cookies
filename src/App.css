html {
    background-image: url(./images/linedpaper.png);
    height: 100%;
}

body {
    height: 100%;
    font-family: 'Nanum Pen Script', cursive;
    font-size: 28px;
}

a {
    color: inherit;
    text-decoration: none;
}

button {
    font-size: 28px;
    font-family: 'Nanum Pen Script', cursive;
}

input {
    font-size: 28px;
    font-family: 'Nanum Pen Script', cursive;
}

header {
    font-family: 'Baloo Thambi 2', cursive;
}

#root {
    height: 100%;
}

.App {
    display: flex;
    flex-direction: column;
    text-align: center;
    height: 100%;
}

.Main {
    flex: 1;
}

.Header {
    background-color: #5D4037;
    color: white;
    font-size: 2.5em;
    padding: 48px;
}

.Cookie {
    animation: Spin-image infinite 20s linear;
    height: 80px;
}

.Box {
    display: inline-block;
    vertical-align: top;
    margin: 6px;
    width: 180px;
    height: 150px;
    padding: 20px;
    color: white;
    border-radius: 16px;
}

.Wrapper {
    width: 100%;
    height: 100%;
    display: table;
}

.BoxesWrapper {
    display: table-cell;
    width: 75%;
    height: 100%;
    padding: 0 12px;
}

.OrderFormWrapper {
    display: table-cell;
    text-align: left;
    width: 25%;
    height: 100%;
    padding-left: 24px;
}

.OrderForm input[type=text] {
    width: 100%;
    padding: 6px;
    margin: 8px 0;
    box-sizing: border-box;
    border-radius: 4px;
}

.BoxLabel {
    font-family: 'Baloo Thambi 2', cursive;
    font-size: 1em;
    margin: 0;
}

.OrderLabel {
    flex: 1;
}

.OrderOption {
    height: 100%;
    width: auto;
    margin: 6px 0;
    padding: 6px;
    border-radius: 4px;
    display: flex;
    align-items: center;
    border: 2px solid #906355;
}

.OrderOptionList {
    margin-top: 12px;
}

@keyframes Spin-image {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}

.Button {
    color: #30211c;
    padding: 6px 12px 6px;
    border-radius: 4px;
    border: 2px solid #5D4037;
}

Button:focus {
    outline: none;
    color: white;
    background-color: #5D4037;
}

Button:hover {
    cursor: pointer;
    color: white;
    background-color: #5D4037;
}

.Footer {
    flex: 0;
    background-color: #5D4037;
    color: white;
    width: 100%;
    padding: 6px 0 6px;
}

.Alert {
    padding: 12px 0 12px;
    text-decoration: underline solid darkred;
}