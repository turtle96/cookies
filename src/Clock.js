// using component allows use of additional features such as local state and lifecycle hooks
import React from "react";

class Clock extends React.Component {
  constructor(props) {
    // call the base constructor with props
    super(props);

    // assigns the initial this.state
    this.state = {date: new Date()};
  }

  ///// lifecycle hooks setup

  // runs after the component output has been rendered to the DOM
  componentDidMount() {
    this.timerID = setInterval(
      () => this.tick(),
      1000
    );
  }

  // called when a component is being removed from the DOM
  componentWillUnmount() {
    // clears the timer
    clearInterval(this.timerID);
  }

  tick() {
    // updates the state when called
    // on setState() call
    // React knows the state has changed, and calls the render() method again
    this.setState({
      date: new Date()
    });

    // do not modify this.state directly i.e. this.state.comment = 'Hello';
    // State Updates May Be Asynchronous
  }

  render() {
    return (
      <div>
        <h2 className="App-intro">It is {this.state.date.toLocaleTimeString()}.</h2>
      </div>
    );
  }
}

export default Clock;