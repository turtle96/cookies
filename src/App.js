// All React components must act like pure functions with respect to their props.

import React, {Component} from 'react';
import './App.css';
import Clock from "./Clock";
import Boxes from "./Box";

class App extends Component {
  render() {
    return (
      <div className="App">
        <Clock/>
        <Order/>
      </div>
    );
  }
}

class Order extends Component {
  constructor(props) {
    super(props);
    this.state = {boxes: []};

    // This binding is necessary to make `this` work in the callback
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.setState({boxes: this.state.boxes.concat("Apple")});
  }

  render() {
    return (
      <div>
        <h1>Ordered: {this.state.boxes.length} {getBoxOrBoxes(this.state.boxes.length)}</h1>
        <Boxes boxes={this.state.boxes}/>
        <button onClick={this.handleClick} className="Button">Order</button>
      </div>
    );
  }
}

function getBoxOrBoxes(length) {
  if (length === 1) {
    return "box";
  } else {
    return "boxes";
  }
}

export default App;
