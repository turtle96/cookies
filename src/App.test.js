import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Clock from "./Clock";
import {mount} from "enzyme";
import Boxes from "./Box";

let originalDateNow;

jest.useFakeTimers();

beforeEach(function () {
  originalDateNow = Date.now;
  Date.now = function () {
    let date = new Date('2017-06-22');
    date.setHours(3);
    date.setMinutes(5);
    date.setSeconds(5);
    return date;
  };
  setInterval.mock.calls = [];
});

afterEach(function () {
  Date.now = originalDateNow;
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('app should render a clock', () => {
  const wrapper = mount(<App />);
  const clock = <Clock/>;
  expect(wrapper.contains(clock)).toEqual(true);
});

it('clock should advance by 1 second', () => {
  const wrapper = mount(<Clock />);

  const now = Date.now();

  // before time skip 1 second
  expect(wrapper.state().date).toEqual(now);

  Date.now = function () {
    let date = new Date(now);
    date.setSeconds(date.getSeconds() + 1);
    return date;
  };

  // run 1 second
  jest.runTimersToTime(1000);

  // called setInterval()
  expect(setInterval.mock.calls.length).toEqual(1);

  now.setSeconds(now.getSeconds() + 1);

  // time went forward 1 second
  expect(wrapper.state().date).toEqual(now);
});

it('render boxes with a colour and correct label', () => {
  const samples = ["Lemon Tea", "Strawberry", "Apple Cinnamon"];
  const wrapper = mount(<Boxes boxes={samples}/>);

  const boxes = wrapper.find('Box');

  expect(boxes.length).toEqual(3);

  boxes.children().forEach(function (item) {
    console.log(item.props().style.backgroundColor);
    expect(item.props().style.backgroundColor).toBeDefined();
  });

  const renderedLabels = boxes.map(function (item) {
    return item.props().label;
  });

  expect(renderedLabels).toEqual(samples);
});

//todo do snapshot testing