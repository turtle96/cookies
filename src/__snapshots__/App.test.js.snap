// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App match snapshot 1`] = `
<div
  className="App"
>
  <div
    className="Main"
  >
    <header
      className="Header"
    >
      Let's Order Cookies
    </header>
    <h2
      className="Standard-text"
    >
      It is 
      03:05:05
      .
    </h2>
    <div>
      <h1
        className="Label"
      >
        Ordered: 
        0
         
        boxes
      </h1>
      <div />
      <form
        onSubmit={[Function]}
      >
        <label>
          Name:
          <input
            name="name"
            onChange={[Function]}
            type="text"
            value={undefined}
          />
        </label>
        <label>
          Address:
          <input
            name="address"
            onChange={[Function]}
            type="text"
            value={undefined}
          />
        </label>
        <input
          type="submit"
          value="Submit"
        />
      </form>
      <div>
        <label>
          Apple
          <button
            className="Button"
            onClick={[Function]}
          >
            Order
          </button>
        </label>
        <label>
          Chocolate Chip
          <button
            className="Button"
            onClick={[Function]}
          >
            Order
          </button>
        </label>
        <label>
          Lemon Barley
          <button
            className="Button"
            onClick={[Function]}
          >
            Order
          </button>
        </label>
        <label>
          Oatmeal
          <button
            className="Button"
            onClick={[Function]}
          >
            Order
          </button>
        </label>
      </div>
    </div>
  </div>
  <footer
    className="Footer"
  >
    Â© 
    2018
  </footer>
</div>
`;
